### PR TITLE
feat(server): Json rdb support

### LINKED ISSUE #587: JSON type should be supported for replication and saves.
**Did you search GitHub Issues and GitHub Discussions First?**
This become an issue after #238 that added the new type JSON.

**Is your feature request related to a problem? Please describe.**
This can become an issue in case a save or replication is made when the dataset contain JSON type.

**Describe the solution you'd like**
Being able to save/replication dataset that contain JSON entries.
For example:
Running the following
```
JSON.SET doc $ '{"a":2}'
```
will add a new entry for key "doc" with the value type being JSON.
This then cannot be replication or save since this type is not part of the RDB type system.
It is not expected that this would become part of RDB types, and would be compatible with REDIS.

**Describe alternatives you've considered**
Since the JSON type is not part of the RDB data types, the only way would be to save this as a string and somehow know that it should then read back as JSON, but this would have the issue that this is not supported as well.

**Additional context**
This would required it own formatting as JSON has more internal types (such as bool, float, string, arrays, decimal and so on) than RDB, and it is not either subset or super set of it.

### PR DESCRIPTION
**Pull request by dranikpg**
Fixes #587 
- Support json inside rdb saver and loader
- Add json values & commands to pytest data generator

### CODE REVIEW

File: tests/dragonfly/utility.py [hunk 1]
```diff
@@ -141,23 +143,35 @@ def rand_str(k=3, s=''):
 
         if t == ValueType.STRING:
             # Random string for MSET
-            return rand_str(self.val_size)
+            return (rand_str(self.val_size),)
         elif t == ValueType.LIST:
             # Random sequence k-letter elements for LPUSH
-            return ' '.join(rand_str() for _ in range(self.val_size//4))
+            return tuple(rand_str() for _ in range(self.val_size//4))
```
Comment (dranikpg) [inline tests/dragonfly/utility.py:149]: I had issues with correctly escaping the string values when formatting cmds, so I switched to tuples (which is more correct)

File: src/server/rdb_save.cc [hunk 2]
```diff
@@ -529,6 +540,11 @@ error_code RdbSerializer::SaveStreamObject(const robj* obj) {
   return error_code{};
 }
 
+error_code RdbSerializer::SaveJsonObject(const PrimeValue& pv) {
+  auto json_string = pv.GetJson()->to_string();
+  return SaveString(json_string);
+}
```
Comment (dranikpg) [inline src/server/rdb_save.cc:546]: In the future we should maybe enable one of jsoncons extensions and use a binary format like bson

File: src/redis/rdb.h [hunk 3]
```diff
@@ -101,7 +101,7 @@
 /* NOTE: WHEN ADDING NEW RDB TYPE, UPDATE rdbIsObjectType() BELOW */
 
 /* Test if a type is an object type. */
-#define rdbIsObjectType(t) ((t >= 0 && t <= 7) || (t >= 9 && t <= 18))
+#define __rdbIsObjectType(t) ((t >= 0 && t <= 7) || (t >= 9 && t <= 18))
```
Comment (dranikpg) [inline src/redis/rdb.h:104]: Should I delete it? Rename it? Keep the name and use something else for the df version?

Comment (romange) [reply src/redis/rdb.h:104]: This is ok. Maybe just add a comment

File: tests/dragonfly/utility.py [hunk 4]
```diff
@@ -64,6 +65,7 @@ class ValueType(Enum):
     SET = 2
     HSET = 3
     ZSET = 4
+    JSON = 5
```
Comment (romange) [inline tests/dragonfly/utility.py:68]: Weird indentation

**Review by romange — APPROVED** [review]

File: tests/dragonfly/utility.py (hunk 4 above)
Comment (dranikpg) [reply tests/dragonfly/utility.py:68]: Those are class fields - pythonic way to make an enum. Better than making it all just global flags

**Pull request merged**